src/frontend/src/service/__mocks__/axios.js

```diff
@@ -0,0 +1,66 @@
+const axios = {
+  request: jest.fn(request => {
+    if (
+      (request.url === 'http://localhost:5000/api/hello') &
+      (request.method === 'GET')
+    ) {
+      return Promise.resolve({
+        data: 'Hello, world!',
+        status: 200,
+        statusText: 'OK'
+      });
+    } else if (
+      (request.url === 'http://localhost:5000/api/search') &
+      (request.method === 'POST')
+    ) {
+      return Promise.resolve({
+        data: [
+          {
+            name: 'WOODS, LAVELLE D',
+            birth_year: 1970,
+            case_number: 'ZA0081909',
+            citation_number: 'ZA0081909',
+            location: 'Multnomah',
+            date: '07/02/2013',
+            violation_type: 'Offense Violation',
+            current_status: 'Inactive',
+            balance_due: '0',
+            charges: [
+              {
+                name: 'Failure to Properly Use Safety Belts - MV Operator',
+                statute: '8112101A',
+                level: 'Violation Class D',
+                date: '06/11/2013',
+                disposition: {
+                  date: '07/16/2013',
+                  ruling: 'Convicted'
+                },
+                expungement_result: {
+                  type_eligibility: 'True',
+                  type_eligibility_reason: 'some string',
+                  time_eligibility: 'True',
+                  time_eligibility_reason: 'some string',
+                  date_of_eligibility: 'date'
+                }
+              }
+            ],
+            case_detail_link:
+              'https://publicaccess.courts.oregon.gov/PublicAccessLogin/CaseDetail.aspx?CaseID=9036658'
+          }
+        ],
+        status: 200,
+        statusText: 'OK'
+      });
+    } else if (
+      request.url.includes(
+        'http://localhost:5000/api/' & (request.method === 'GET')
+      )
+    ) {
+      return Promise.reject(new Error('Request failed with status code 404'));
+    } else if (!request.url.includes('http://localhost:5000/api/')) {
+      return Promise.reject(new Error('bad url'));
+    }
```

Comment:
Should we have a final `else` that throws an exception to guard against typos in the URL when people are writing tests? This is a small addition that could save someone a lot of frustrating debugging.

Reply:
I updated the error message to provide the correct base url incase there is a typo.

Reply:
I made a default error message incase none of the "else if's" wont get triggered.